### PR TITLE
only producer can grant reward

### PR DESCRIPTION
**Pull request by CoderZhi**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] Test A
- [] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: action/protocol/rewarding/protocol.go [hunk 1]
```diff
@@ -183,6 +183,16 @@ func createGrantRewardAction(rewardType int, height uint64) action.Envelope {
...
+func (p *Protocol) Validate(ctx context.Context, act action.Action, sr protocol.StateReader) error {
+	switch act.(type) {
+	case *action.GrantReward:
+		if !address.Equal(protocol.MustGetBlockCtx(ctx).Producer, protocol.MustGetActionCtx(ctx).Caller) {
+			return errors.New("Only producer could create reward")
+		}
+	}
+	return nil
+}
+
```
Comment (dustinxie) [inline action/protocol/rewarding/protocol.go:195]: as chatted, guard with Newfoundland height, can revert after HF activation

**Review by Liuhaai — APPROVED** [review]

**Review by dustinxie — APPROVED** [review]

**Review by Newman1515 — APPROVED** [review]
Ok

**Comment by Newman1515** [comment]
https://github.com/iotexproject/iotex-core/blob/master/.circleci/circleci_github_release.sh

File: action/protocol/rewarding/protocol.go (hunk 1 above)
Comment (Newman1515) [reply action/protocol/rewarding/protocol.go:203]: https://github.com/iotexproject/iotex-core/blob/master/.circleci/circleci_github_release.sh

**Review by Newman1515 — APPROVED** [review]

**Comment by Newman1515** [comment]
> # Description
> 
> Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.
> 
> Fixes # (issue)
> 
> ## Type of change
> 
> Please delete options that are not relevant.
> - [] Bug fix (non-breaking change which fixes an issue)
> 
> # How Has This Been Tested?
> 
> Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
> - [] Test A
> - [] Test B
> 
> **Test Configuration**:
> 
> - Firmware version:
> - Hardware:
> - Toolchain:
> - SDK:
> 
> # Checklist:
> - [] My code follows the style guidelines of this project
> - [] I have performed a self-review of my code
> - [] I have commented my code, particularly in hard-to-understand areas
> - [] I have made corresponding changes to the documentation
> - [] My changes generate no new warnings
> - [] I have added tests that prove my fix is effective or that my feature works
> - [] New and existing unit tests pass locally with my changes
> - [] Any dependent changes have been merged and published in downstream modules
>

**Pull request merged**